src/value_parser.zig

```diff
@@ -83,9 +87,10 @@ fn stringData(comptime DestinationType: type) ValueData {
     return .{
         .value_size = @sizeOf(DestinationType),
         .value_parser = struct {
-            fn parser(dest: *anyopaque, value: []const u8) ValueParseError!void {
+            fn parser(dest: *anyopaque, value: []const u8, alloc: Allocator) ValueParseError!void {
                 const dt: *DestinationType = @ptrCast(@alignCast(dest));
-                dt.* = value;
+                const cpy = try alloc.dupe(u8, value);
```

Comment:
@Dinastyh thanks for your contribution. I apologize for the long silence from my side.

Two questions to this change:

1. Why do we need to copy the value? Aren't the command line arguments available during the entire program run?
2. Since the string parser the only value parser that requires the allocator, shouldn't the allocator be passed to the value parser itself, so that there is no need to pass the allocator to every `parse` invocaton?

Reply:
Hi,
No problem don't worry.

1. No they are allocated by std.process.argsWithAllocator (file: app_runner.zig, function getAction, line: 48), this iterator is initalized with the Apprunner Arena Allocator. This arena allocator is deinit on succes in the same function (file: app_runner.zig, function getAction, line: 55)
2. I wanted to keep homogeneous parser function prototype for each type but it's also an option